### PR TITLE
NPCs and improved Toasts

### PR DESCRIPTION
**Pull request by thequeenofspades**
NPCs can now be added to areas in much the same way as Placeables and Clickables.
Clicking on an NPC triggers a conversation with that NPC. Their avatar will switch to the conversation avatar size and position.
Ending a conversation now refreshes the display of NPCs in the area (so that the dialoguing NPC will go back to where they were).
Mousing over a Clickable/Placeable/NPC displays a brief Toast with their name. This Toast disappears when the player stops mousing over, instead of after the usual 5-second delay.
Toasts now replace previous Toasts instead of rendering over them.
image

### CODE REVIEW

File: src/objects/ConversationManager.js [hunk 1]
```diff
@@ -153,9 +163,16 @@ ConversationManager.prototype.endConversation = function() {
...
+  for (var i = 0; i < this._game.room.npcs.length; i++) {
+    var npc = this._game.room.npcs[i];
+    npc.show();
+  }
+
   if ('onEnd' in this.conversation) {
     this.customActions.customAction(this.conversation['onEnd']);
   }
+
+  this.shown = [];
```
Comment (drcaramelsyrup) [inline src/objects/ConversationManager.js:175]: Oh thank goodness. Also a great way to gray out options if we want.

File: src/objects/Toast.js [hunk 2]
```diff
@@ -30,6 +35,8 @@ function Toast(game, message, timeout) {
   this.toast.displayObject.width = this.toastWidth;
   this.toast.displayObject.height = this.toastHeight;
 
+  game.toast = this;
```
Comment (drcaramelsyrup) [inline src/objects/Toast.js:38]: Is this for a "global" toast object?

**Review by drcaramelsyrup — APPROVED** [review]
A few clarification comments but LGTM +1

File: src/objects/Toast.js (hunk 2 above)
Comment (thequeenofspades) [reply src/objects/Toast.js:38]: Yep! So that you can make them go away every time a new Toast appears.

**Pull request merged**